hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstWalker.java

```diff
@@ -226,6 +228,22 @@ public void visitQuerySpec(QuerySpec querySpec) {
 
 	@Override
 	public void visitSortSpecification(SortSpecification sortSpecification) {
+		NullPrecedence nullPrecedence = sortSpecification.getNullPrecedence();
+		final boolean hasNullPrecedence = nullPrecedence != null && nullPrecedence != NullPrecedence.NONE;
+		if ( hasNullPrecedence && ! dialect.supportsNullPrecedence() ) {
+			appendSql( "case when (" );
```

Comment:
This should be avoided when the null precedence matches the DBMS default.

Reply:
I have introduced a new method to `Dialect`, i.e. `supportsNullPrecedence()`. The above code is only run when the DBMS doesn't support it.
Am I missing something?

Reply:
DBMS have a default null precedence in which case you don't need to render the nulls clause or the emulation. Here is an example for MySQL: https://gregrs-uk.github.io/2011-02-02/mysql-order-by-with-nulls-first-or-last/
Another example from the PostgreSQL documentation `By default, null values sort as if larger than any non-null value; that is, NULLS FIRST is the default for DESC order, and NULLS LAST otherwise.`: https://www.postgresql.org/docs/8.3/queries-order.html

Reply:
Another excellent PR comment. It might be a good optimization effort, but given the current maturity status of v6, I wanna keep it out of the scope of this PR and might consider it in the future.